src/WorkerChannel.ts

```diff
@@ -74,18 +82,60 @@ export class WorkerChannel {
         }
     }
 
+    getBaseHookContext(functionInvocationId?: string, msgCategory?: string): HookContext {
+        const log = (logLevel: LogLevel, ...args: any[]) =>
+            this.#userLog(logLevel, functionInvocationId, msgCategory, ...args);
+
+        const logger = Object.assign((...args: any[]) => log(LogLevel.Information, ...args), {
+            info: (...args: any[]) => log(LogLevel.Information, ...args),
+            warn: (...args: any[]) => log(LogLevel.Warning, ...args),
+            error: (...args: any[]) => log(LogLevel.Error, ...args),
+            verbose: (...args: any[]) => log(LogLevel.Trace, ...args),
+        });
+
+        return {
+            hookData: this.#hookData,
+            logger,
+        };
+    }
+
+    #userLog(level: LogLevel, functionInvocationId?: string, msgCategory?: string, ...args: any[]): void {

```

Comment:
> Added a logger property to HookContext, which allows all hooks, including invocation and app startup hooks, to emit user logs to app insights.

To me, the cons outweigh the pros on this one. I think https://github.com/Azure/azure-functions-host/issues/8222 is a blocker and I want to keep the core api as simple as possible until we make more progress on the programming model. We can have a discussion on this and perhaps you can convince me otherwise, but I would prefer if we do that in a separate PR as I think this would be a feature on its own

Reply:
Yeah makes sense. I've removed it for now